pypuppetdb/types.py

```diff
@@ -128,22 +176,36 @@ def __str__(self):
     def __unicode__(self):
         return self.__string
 
+    def events(self):
+        """Get all events for this report."""
+        return self.__api.events(query=self.__query_scope)
```

Comment:
I'll fix that the next time I get a chance. 

As another improvement I may as well accept **kwargs in these functions. Much greater flexibility, especially with `self.__api.reports(query=self.__query_scope)`